src/raft.rs

```diff
@@ -2214,6 +2175,25 @@ impl<T: Storage> Raft<T> {
         if self.maybe_commit() {
             self.bcast_append();
         }
+
+        // The quorum size is now smaller, consider to response some read requests.
+        // If there is only one peer, all pending read requests must be responsed.
+        if let Some(ctx) = self.read_only.last_pending_request_ctx() {
```

Comment:
What if some reads in the front of the queue are ready to be processed?

Reply:
Without iterating the pending queue, they can only become ready when more `heartbeat responses` are received. Do you have any solution to handle them without iterating?